x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java

```diff
@@ -74,6 +75,11 @@ public ClassificationInferenceResults(StreamInput in) throws IOException {
         } else {
             this.predictionFieldType = PredictionFieldType.STRING;
         }
+        if (in.getVersion().onOrAfter(Version.V_7_9_0)) {
```

Comment:
`V_7_10_0` surely?

Reply:
potentially, I am not sure if we want to backport it to 7.9 as java side inference was introduced in 7.9

Reply:
@droberts195 what do you think?

Reply:
If it means the data in the destination index would have inconsistent fields without this fix in 7.9 when the fields were consistent in 7.8 then this is fixing a regression and should go into 7.9.